nixpkgs/nixpkgs.bzl

```diff
@@ -106,21 +107,46 @@ def _nixpkgs_package_impl(ctx):
   _symlink_children(output_path, ctx)
 
 
-nixpkgs_package = repository_rule(
-  implementation = _nixpkgs_package_impl,
+nix_call = repository_rule(
+  implementation = _nix_call_impl,
   attrs = {
+    "executable_name": attr.string(),
+    "extra_arguments": attr.string_list(),
     "attribute_path": attr.string(),
     "nix_file": attr.label(allow_single_file = [".nix"]),
     "nix_file_deps": attr.label_list(),
     "nix_file_content": attr.string(),
     "path": attr.string(),
+    "repositories": attr.label_keyed_string_dict(),
     "repository": attr.label(),
     "build_file": attr.label(),
     "build_file_content": attr.string(),
   },
   local = True,
 )
 
+def nixpkgs_package(*args, **kwargs):
+    nix_call(
+        executable_name = "nix-build",
+        extra_arguments = [
+          # Creating an out link prevents nix from garbage collecting the store
+          # path.  nixpkgs uses `nix-support/` for such house-keeping files, so
+          # we mirror them and use `bazel-support/`, under the assumption that
+          # no nix package has a file named `bazel-support` in its root.  A
+          # `bazel clean` deletes the symlink and thus nix is free to garbage
+          # collect the store path.
+          "--out-link", "bazel-support/nix-out-link"
+        ],
+        *args,
+        **kwargs
+    )
+
+def nix_instantiate(*args, **kwargs):
```

Comment:
Is this function used anywhere? It looks like dead code to me.

Reply:
Ouch sorry, pushed the wrong commit :/

Should be fixed now